### PR TITLE
add request latency tracking

### PR DESCRIPTION
**Pull request by mjudeikis**
Adds simple request latency tracking to requests in our blackbox monitoring.
```release-note
NONE
```

### CODE REVIEW

**Comment by mjudeikis** [comment]
@jim-minter, 
At the moment if you query in the `fakerp-upgrades-insights` there is not much data to work with.
Artefacts data in the PR does not give much information at the moment too. 
This will give us some start and ability to play with data. 
```
requests
| where timestamp > ago(30m) 
| summarize avg(duration) by name, id, duration
```

**Comment by mjudeikis** [comment]
> yikes that new dependency is pulling in a lot of stuff!

I know! sad :/

File: cmd/monitoring/monitoring.go [hunk 1]
```diff
@@ -93,7 +100,7 @@ func (m *monitor) listResourceGroupMonitoringHostnames(ctx context.Context, subs
 	// these routes will not be available at creation time, so we will not check for their availability
 	oc, err := loadOCConfig()
 	if err != nil {
-		m.log.Warn("failed to load OpenShiftManagedCluster config, will not monitor routes")
+		m.log.Warnf("failed to load OpenShiftManagedCluster config, will not monitor routes. Error %s", err.Error())
```
Comment (charlesakalugwu) [inline cmd/monitoring/monitoring.go]: m.log.Warnf("failed to load OpenShiftManagedCluster config, will not monitor routes: %v", err)

File: pkg/util/blackbox/types.go [hunk 2]
```diff
@@ -66,11 +70,23 @@ func (c *Config) Start(ctx context.Context) {
...
 			start := time.Now()
-
 			resp, err := c.Cli.Do(c.Req)
-
 			end := time.Now()
 
+			if c.Icli != nil {
+				if resp != nil {
+					request := appinsights.NewRequestTelemetry(c.Req.Method, c.Req.URL.String(), time.Second, resp.Status)
+					request.Id = os.Getenv("RESOURCEGROUP")
```
Comment (charlesakalugwu) [inline pkg/util/blackbox/types.go]: so all requests will have the same value (cluster resource group) for a given cluster?

Comment (mjudeikis) [reply pkg/util/blackbox/types.go]: Yes, its like identifier. So our upgrade PR cluster names are already unique. You can play with data in the app-insights instance in our sub already.

**Comment by mjudeikis** [comment]
stripped the dependencies as we have all of the already and it was just updating them.

**Comment by asalkeld** [comment]
tests need passing, but lgtm

**Comment by jim-minter** [comment]
sure there's going to be a million changes to the monitoring schema but it's a start :)

**Comment by mjudeikis** [comment]
https://github.com/openshift/openshift-azure/issues/1362 
/retest

**Comment by mjudeikis** [comment]
https://github.com/openshift/openshift-azure/issues/1362
/retest

**Comment by mjudeikis** [comment]
E2E failed with disk being attached after 15 min, when health check failed already
Scaledown, e2e-upgrade random CI flake
/retest

**Comment by mjudeikis** [comment]
Just get in you...
/retest

**Pull request merged**